### ISSUE TITLE
Package upload APIs should be included in swagger docs

### ISSUE THREAD

**Issue by Infinoid** [go-gitea/gitea#30597]
### Description

There are many API endpoints for uploading packages of various types.  However, these endpoints are not described in the API swagger data, so they are not visible in the swagger UI.

Example: `PUT https://gitea.example.com/api/packages/{owner}/debian/pool/{distribution}/{component}/upload` is the endpoint for uploading a .deb package to an APT repository.  It is described [here](https://docs.gitea.com/next/usage/packages/debian#publish-a-package) on docs.gitea.com, but not [here](https://try.gitea.io/api/swagger#/package) in the swagger docs.

Other examples include [conda](https://docs.gitea.com/next/usage/packages/conda#publish-a-package), [generic](https://docs.gitea.com/next/usage/packages/generic#publish-a-package), [go](https://docs.gitea.com/next/usage/packages/go#publish-a-package).

Are these API endpoints treated differently because they exist under `/api/packages/` rather than `/api/v1/packages/`?  If so, can they be aliased?

### Gitea Version

1.21.10, 1.23.0+dev-172-geb24d973b0

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

![image](https://github.com/go-gitea/gitea/assets/50254/e106c8b1-a50e-4405-bc01-465b5df0447d)

### How are you running Gitea?

container image, and try.gitea.io

### Database

None

**Comment by lunny** [comment]
But the protocol is this package's own standard. Should we list them?

**Comment by KN4CK3R** [comment]
It's not mentioned in swagger because what lunny says. But all packages which use `curl` in the examples may benefit from swagger because the package type does not have a buildin upload.

**Comment by Infinoid** [comment]
Yeah, you're both right.  I'm thinking of the ones that use `curl`.  The package types which have official and well-defined clients, like `docker`, don't need explicit documentation as much.

**Comment by yp05327** [comment]
I think this is a `proposal` not a `bug`
